pkg/apis/cr/v1alpha1/types.go

```diff
@@ -253,49 +290,73 @@ const (
 
 // Location
 type Location struct {
-	Type     LocationType `json:"type"`
-	Bucket   string       `json:"bucket"`
-	Endpoint string       `json:"endpoint"`
-	Prefix   string       `json:"prefix"`
-	Region   string       `json:"region"`
+	// Type specifies the kind of object storage that would be used to upload the backup objects. Currently supported values are: "GCS", "S3Compliant", "Azure" and "Kopia".
+	Type LocationType `json:"type"`
+	// Bucket represents the bucket on the object storage where the backup would be uploaded to.
+	Bucket string `json:"bucket"`
+	// Endpoint specifies the endpoint where the object storage is accessible at.
+	Endpoint string `json:"endpoint"`
+	// Prefix is the string that would be prepended into the object path in the bucket where the objects will be uploaded to.
+	Prefix string `json:"prefix"`
+	// Region represents the region where the bucket mentioned above is present in.
+	Region string `json:"region"`
 }
 
 // CredentialType
 type CredentialType string
 
 const (
+	// CredentialTypeKeyPair represents the Credential being used is of Type key pair.
 	CredentialTypeKeyPair CredentialType = "keyPair"
-	CredentialTypeSecret  CredentialType = "secret"
-	CredentialTypeKopia   CredentialType = "kopia"
+	// CredentialTypeSecret represents the Credential being used is of Type Secret.
+	CredentialTypeSecret CredentialType = "secret"
+	// CredentialTypeKopia represents the Credential being used is of Type Kopia.
+	CredentialTypeKopia CredentialType = "kopia"
 )
 
 // Credential
 type Credential struct {
-	Type              CredentialType     `json:"type"`
-	KeyPair           *KeyPair           `json:"keyPair,omitempty"`
-	Secret            *ObjectReference   `json:"secret,omitempty"`
+	// Type represents the information about how the credentials are going to be provided for the respective object storage.
+	Type CredentialType `json:"type"`
+	// KeyPair represnets the pair of Keys being used for the Credential of Type KeyPair.
+	KeyPair *KeyPair `json:"keyPair,omitempty"`
+	// Secret represents the Kubernetes Secret Object used for the Credential of Type Secret.
+	Secret *ObjectReference `json:"secret,omitempty"`
+	// KopiaServerSecret represents the secret being used by Credential of Type Kopia.
 	KopiaServerSecret *KopiaServerSecret `json:"kopiaServerSecret,omitempty"`
 }
 
 // KeyPair
 type KeyPair struct {
-	IDField     string          `json:"idField"`
-	SecretField string          `json:"secretField"`
-	Secret      ObjectReference `json:"secret"`
+	// IDField represents the IDs of corresponding keys in the Secret under which KeyPair
```

Comment:
```suggestion
	// IDField represents the IDs of corresponding keys in the Secret under which KeyPair
```
IDs or ID?

Reply:
Since this field can be used to store IDs of multiple keys, so maybe IDs is correct?